powerline_shell/__init__.py

```diff
@@ -188,6 +188,8 @@ def main():
     arg_parser = argparse.ArgumentParser()
     arg_parser.add_argument('--generate-config', action='store_true',
                             help='Generate the default config and print it to stdout')
+    arg_parser.add_argument('--config-file', action='store', default='~/.powerline-shell.json',
```

Comment:
I think just `--config` would be the more common way to name this option. You could also give it the `-c` shorthand.

Reply:
Also the default being `~/.powerline-shell.json` is not quite correct. There are two issues:

- The preferred default is actually the `os.path.join(os.environ.get("XDG_CONFIG_HOME", "~/.config"), "powerline-shell", "config.json")` file - ~/.powerline-shell.json is going to be removed eventually.
- I think it would be better to remove `default` entirely from this option. If someone were to do `powerline-shell --help` it would say the default location is `~/.powerline-shell.json`, but if they happened to have a file called `powerline-shell.json` in their current directory, then the default value would not actually be used. This confusing behavior.

Reply:
Good points. It occurs to me also that if you specify a config file, it should confirm that the exact path you supplied works. It shouldn't silently ignore a --config path that doesn't exist if the user explicitly passed it.

I updated the patch with your changes, and also, made it so that adding a -c option just skips the list of search paths altogether. Let me know your thoughts.